rmf_task/include/rmf_task/requests/ChargeBatteryFactory.hpp

```diff
@@ -35,7 +38,7 @@ class ChargeBatteryFactory : public RequestFactory
 {
 public:
 
-  ChargeBatteryFactory();
+  ChargeBatteryFactory(std::optional<std::string> requester = std::nullopt);
```

Comment:
Hmm, since this is derived from the request factory abstraction, are we good to break API to add those fields into `make_request`?

Another possibility for request time would be to make these constructors accept a `std::function` that returns `rmf_traffic::Time`. Let me know which sounds better

Reply:
Something like this https://github.com/open-rmf/rmf_task/pull/81/commits/da239d70794deb6d6b3f8b57717b5b4f7c790ab1?